Add dummy _dl_random.

---
 dl-sysdep.c |    1 +
 1 file changed, 1 insertion(+)

--- glibc/sysdeps/mach/hurd/dl-sysdep.c.orig	2009-07-29 20:34:05.000000000 +0000
+++ glibc/sysdeps/mach/hurd/dl-sysdep.c	2009-07-29 20:33:51.000000000 +0000
@@ -56,6 +56,7 @@
 				   of init-first.  */
 /* This variable containts the lowest stack address ever used.  */
 void *__libc_stack_end;
+void *_dl_random attribute_relro = NULL; /* TODO */
 
 #if HP_TIMING_AVAIL
 hp_timing_t _dl_cpuclock_offset;
